utils.py

```diff
@@ -80,25 +81,33 @@ def date_format(text: str) -> str:
     return f"{dd}/{mm}/{yyyy}"
 
 
-def has_role(role: str, user: str) -> bool:
+def has_role(role: Role, user: str, strict_voice_check: bool = False) -> bool:
     """Checks if a user has a PS role or higher.
 
     Args:
-        role (str): PS role (i.e. "voice", "driver").
+        role (Role): PS role (i.e. "voice", "driver").
         user (str): User to check.
 
-    Raises:
-        BaseException: role is unrecognized.
-
     Returns:
         bool: True if user meets the required criteria.
     """
-    roles = {"voice": "+*%@★#&~", "driver": "%@#&~"}
-    if role in roles:
-        if user and user[0] in roles[role]:
+    roles: Dict[Role, str] = {
+        "admin": "~&",
+        "owner": "~&#",
+        "bot": "*",
+        "host": "★",
+        "mod": "~&#@",
+        "driver": "~&#@%",
+        "player": "☆",
+        "voice": "~&#@%+",
+        "prizewinner": "^",
+    }
+    if user:
+        if user[0] in roles[role]:
+            return True
+        if role == "voice" and not strict_voice_check and user[0] not in "*★☆^":
```

Comment:
Ho due cose da osservare con `strict_voice_check == False`.

" " è il rank nativo associato ai regular user e va quindi aggiunto ai simboli da escludere.

Vanno esclusi anche i caratteri alfanumerici, perché `has_role` viene invocato anche dall'handler di `|queryresponse|userdetails|` per parsare il dizionario `rooms`:
```
|queryresponse|userdetails|{"id":"armastrong","userid":"armastrong","name":"ArmAstrong","avatar":101,"group":" ","autoconfirmed":false,"status":"","rooms":{"italiano":{},"☆battle-gen5randombattle-1204635570":{"p1":" NOMYLOVE","p2":" ArmAstrong"}}}
```
A differenza di quanto succede per i log della chat, se un utente non è auth in una room non viene lasciato uno spazio prima del roomid e quindi `user[0]` diventa la prima lettera del roomid!

Questa è una possibile modifica:
```suggestion
        if (
            role == "voice"
            and not strict_voice_check
            and user[0] not in " *★☆^"
            and not user[0].isalpha()
        ):
```